### PR TITLE
feat(CORE-1255): Add docs on what pipelines is and how it works

### PR DESCRIPTION
**Pull request by MoonMoon1919**
Adds docs explaining the execution flow and architecture of pipelines v2

### CODE REVIEW

**Comment by josh-padnick** [comment]
@MoonMoon1919 I had...a lot of pent-up docs in me so I used your excellent initial pass to expand on this in a few ways:

- Wrote up more of the "why" in "What is Gruntwork Pipelines?"
- Added a "Core concepts" page
- Repurposed "How it works" into a page dedicated to explaining the architecture
- Added a placeholder for "How it works" intended to describe the how the _UX_ works
- Added pages on "Supported CI systems" and "Supported clouds" to make explicit what we support
- Added a placeholder for "Initial setup"
- Split out the "Architecture" page into the "Architecture" page, "Access control" (your permissions header), and a "Usage data" page
- Added a "Pipelines actions" page

In my opinion, once we fill in the two placeholder sections (How it works, Initial setup), this is ready to ship. I'll also leave some individual comments on a few specific items.

File: _docs-sources/pipelines/architecture/index.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Architecture
+
+## Overview
+
+There are three major components in the Gruntwork Pipelines architecture:
+
+1. The orchestrator
+2. The dispatcher
+3. The executor
```
Comment (josh-padnick) [inline _docs-sources/pipelines/architecture/index.md:9]: Could we simplify our architecture to just the orchestrator and the executor? For example, if the orchestrator does the dispatching to the executor, do we need to call out the dispatcher as a separate component?

File: _docs-sources/pipelines/core-concepts/index.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+Sometimes users create a Pull Request that changes more than one file at a time. And sometimes a change to a single file affects how multiple other files will be "applied" to your live cloud account. For example, many users of Terragrunt use a pattern known as "envcommon" as a way to specify a default set of values for modules.
+
+When more than one infra-change is made at a time, we call this an _infrastructure-change set._ We sometimes call this an "infra-change set" for short.
+
+### Pipelines actions
+
+When a user proposes to make an infra-change by opening a Pull Request, we want to take some kind of "action" in response to that proposed-change. For example, we may want to run a `terragrunt plan` and estimate the cost of applying this Terragrunt plan. We call these _pipelines actions._
+
+Gruntwork is responsible for adding support for a growing library of Pipelines Actions and we will continue to add more over time.
```
Comment (josh-padnick) [inline _docs-sources/pipelines/core-concepts/index.md:27]: I realize we have a very limited number of "pipelines actions" at this point, but the concept that there are many different types of actions we'll support and today we only support a limited number is an important messaging point in sales and key point of shared understanding in product planning.

File: _docs-sources/pipelines/how-it-works/index.md [hunk 3]
```diff
@@ -1,7 +1,9 @@
...
+TODO: Translate this into prose, or headers
 
-## Usage Data
-
-Gruntwork Pipelines captures usage data to better understand how our customers use our products. This information includes the duration of pipelines runs, the number of jobs you run, customer name, and application errors that occur during execution. If you would like to disable telemetry, please set the environment variable `PIPELINES_TELEMETRY_OPT_OUT=1` in your CI job.
+- User opens a Pull Request
+- Pipelines identifies all infra-changes being proposed
+- For each infra-change, Pipelines runs the relevant Pipelines Actions
+- The results of these actions show up as commnets and check runs in the GitHub Pull Request
+- [Future] Configure Pipelines to run different Pipelines Actions.
```
Comment (josh-padnick) [inline _docs-sources/pipelines/how-it-works/index.md:9]: We should aim for something basic, simple, and with a screenshot here so users can get it right away without having to install pipelines themselves.

File: _docs-sources/pipelines/architecture/index.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+### The executor
+
+The executor executes an ordered list of pipelines actions against a specific pull request or git commit in response to a specific infrastructure. It then posts the status and output of these pipelines actions to the pull request, or takes other actions to ensure that the user has visibility into what happened.
+
+## Execution Flow
+
+You can see these components in action below:
+
+![Gruntwork Pipelines Execution Flow](/img/pipelines/how-it-works/pipelines_execution_flow.png)
+
```
Comment (josh-padnick) [inline _docs-sources/pipelines/architecture/index.md:28]: The diagram is helpful in quickly understanding how it all works. My main feedback is around using consistent terminology and embracing Pipelines Actions as a first-class concept (over mere terragrunt plan/apply/destroy).

In that spirit, could we change:
- "Calculate infra-change-set from changes" to "Parse infra-change-set from changes"
- "For each infra change in infra change set" to "For each infra-change in infra-change set"
- "Determine change type" to "Determine infra-change type"
- "Determine terragrunt command" to "Determine Pipelines Actions"
- "Run terragrunt plan/apply/destroy" to "Run Pipelines Actions

File: _docs-sources/pipelines/initial-setup/index.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+- In the infra-live repo:
+  - Create the following GitHub Actions
+- Now make your first change in the infra-live repo
+  - You should see Pipelines run pipelines actions
+- To configure these actions, do xxx...
+
+Alternatively:
+
+- For now, Gruntwork configures Pipelines for you.
+- We'll publish official instructions on how to set it up shortly.
```
Comment (josh-padnick) [inline _docs-sources/pipelines/initial-setup/index.md:20]: This section is key because now new customers can "just set it up" once they get access!

File: sidebars/pipelines.js [hunk 6]
```diff
@@ -44,10 +64,33 @@ const sidebar = [
...
+        items: [
+          {
+            label: "Usage Data",
+            type: "doc",
+            id: "pipelines/usage-data/index",
+          },
+        ],
       },
       {
         label: "Previous Versions",
```
Comment (josh-padnick) [inline sidebars/pipelines.js:96]: Could we somehow include the "external link" icon like we have on Knowledge Base to make clear the user will jump to a different section here? Alternatively, we could make this a "standard" page and the link out to the ECS Deploy Runner pipelines from there?

image

Comment (josh-padnick) [inline sidebars/pipelines.js:96]: The current knowledge base link goes to https://github.com/orgs/gruntwork-io/discussions?discussions_q=label%3As%3ACI%2FPipelines%20sort%3Atop, which will be tags for the _legacy_ pipelines solution ECS Deploy Runner. 

@ebeneliason Could we get a story to update the knowledge base tags as follows:
- Switch `s:CI/Pipelines` to `s:ECS Deploy Runner`.
- Add a new tag `s:Pipelines`.

**Review by josh-padnick — CHANGES_REQUESTED** [review]

**Comment by ellisonc** [comment]
@MoonMoon1919 was this meant to be compared against the devops-foundations branch? @josh-padnick we're working off a branch other than main which causes previews to not work.

File: _docs-sources/pipelines/architecture/index.md (hunk 1 above)
Comment (ellisonc) [reply _docs-sources/pipelines/architecture/index.md:9]: Maybe, but for now we want to describe what exists.

File: _docs-sources/pipelines/core-concepts/index.md (hunk 2 above)
Comment (ellisonc) [reply _docs-sources/pipelines/core-concepts/index.md:27]: We should avoid using future features (like cost estimation) as examples in our documentation. These are fine to call out as potential improvements in the marketing site, but the docs should describe the product as it exists

File: _docs-sources/pipelines/architecture/index.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+There are three major components in the Gruntwork Pipelines architecture:
+
+1. The orchestrator
+2. The dispatcher
+3. The executor
+
+### The orchestrator
+
+The orchestrator identifies each [infrastructure change](../core-concepts/#infrastructure-change) in the Pull Request or git commit, classifies the type of infrastructure change it is (e.g. NewAwsAccount vs. OpenTfModuleUpdate) and determines the right [pipelines actions](../core-concepts/#pipelines-actions) to run based on that infrastructure change. For each infrastructure change, it calls the dispatcher.
```
Comment (ellisonc) [inline _docs-sources/pipelines/architecture/index.md:13]:
```suggestion
The orchestrator identifies each [infrastructure change](../core-concepts/#infrastructure-change) in the Pull Request or git commit, classifies the type of infrastructure change it is (e.g. NewAwsAccount vs. TerraformModuleUpdate) and determines the right [pipelines actions](../core-concepts/#pipelines-actions) to run based on that infrastructure change. For each infrastructure change, it calls the dispatcher.
```
We should avoid calling out open tf until it is available as a feature of pipelines

File: _docs-sources/pipelines/core-concepts/index.md [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+
+## Overview
+
+Fundamentally, Gruntwork Pipelines is designed to roll out an _infrastructure change._ Doing that in a secure and responsible way requires understanding a few key concepts.
+
+## Concepts
+
+### Infrastructure change
+
+When you edit any infrastructure-as-code that needs to be "applied" to your cloud account (e.g. AWS or GCP), you are making an _infrastructure change_. We sometime call these an "infra-change" for short.
```
Comment (ellisonc) [inline _docs-sources/pipelines/core-concepts/index.md:11]:
```suggestion
When you edit any infrastructure-as-code that needs to be "applied" to your cloud account, you are making an _infrastructure change_. We sometime call these an "infra-change" for short.
```
GCP is not supported

File: _docs-sources/pipelines/core-concepts/index.md [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+
+Infra-changes can involve updates to OpenTF (Terraform) code, Packer templates, Docker files, Kubernetes Helm charts, or any other type of file that represents a desired state of your infrastructure and that needs to be somehow applied. A classic example is changing some variables on an instance of a Terraform module. By changing these variables you are essentially requesting that some API calls eventually be made to AWS to update your infrastructure in some way (in this case, by running `terraform apply` at some point).
```
Comment (ellisonc) [inline _docs-sources/pipelines/core-concepts/index.md:13]:
```suggestion
Infra-changes can involve updates to Terraform code, Packer templates, Docker files, Kubernetes Helm charts, or any other type of file that represents a desired state of your infrastructure and that needs to be somehow applied. A classic example is changing some variables on an instance of a Terraform module. By changing these variables you are essentially requesting that some API calls eventually be made to AWS to update your infrastructure in some way (in this case, by running `terraform apply` at some point).
```

File: _docs-sources/pipelines/core-concepts/index.md [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+
+### Infrastructure change set
+
+Sometimes users create a Pull Request that changes more than one file at a time. And sometimes a change to a single file affects how multiple other files will be "applied" to your live cloud account. For example, many users of Terragrunt use a pattern known as "envcommon" as a way to specify a default set of values for modules.
+
+When more than one infra-change is made at a time, we call this an _infrastructure-change set._ We sometimes call this an "infra-change set" for short.
+
+### Pipelines actions
+
+When a user proposes to make an infra-change by opening a Pull Request, we want to take some kind of "action" in response to that proposed-change. For example, we may want to run a `terragrunt plan` and estimate the cost of applying this Terragrunt plan. We call these _pipelines actions._
```
Comment (ellisonc) [inline _docs-sources/pipelines/core-concepts/index.md:25]:
```suggestion
When a user proposes to make an infra-change by opening a Pull Request, we want to take some kind of "action" in response to that proposed-change. For example, we may want to run a `terragrunt plan`. We call these _pipelines actions._
```

File: _docs-sources/pipelines/pipelines-actions/index.md [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+
+## Terragrunt plan
+
+When the pull request is created, Pipelines will automatically execute `terragrunt plan` on every infrastructure-change. The output will be written to a Pull Request comment.
+
+## Terragrunt apply/destroy
+
+When the pull request is merged, Pipelines will automatically execute either `terragrunt apply` or `terragrunt destroy` on every infrastructure change, depending on the type of infrastructure change. For example, if the pull request deletes a `terragrunt.hcl` file, Pipelines will run `terragrunt destroy`.
+
+## Coming soon
```
Comment (ellisonc) [inline _docs-sources/pipelines/pipelines-actions/index.md:13]: I'd remove this section entirely or move its content to the marketing site

File: _docs-sources/pipelines/supported-ci-systems/index.md [hunk 12]
```diff
@@ -0,0 +1,5 @@
+# Supported CI systems
+
+Today, Gruntwork Pipelines works exclusively with [GitHub Actions](https://docs.github.com/en/actions).
+
+However, the [Gruntwork Pipelines architecture](../how-it-works) is designed to be agnostic to the underlying CI system, so we will add support for other CI systems in the future.
```
Comment (ellisonc) [inline _docs-sources/pipelines/supported-ci-systems/index.md:5]: This isn't technically true

File: _docs-sources/pipelines/supported-clouds/index.md [hunk 13]
```diff
@@ -0,0 +1,5 @@
+# Supported clouds
+
+Today, Gruntwork Pipelines works exclusively with Amazon Web Services (AWS).
+
+However, the [Gruntwork Pipelines architecture](../how-it-works) is designed to support any cloud, so we will add support for other CI systems in the future.
```
Comment (ellisonc) [inline _docs-sources/pipelines/supported-clouds/index.md:5]: This also isn't technically true, I'd remove this page

File: _docs-sources/pipelines/access-control/index.md [hunk 14]
```diff
@@ -0,0 +1,23 @@
+# Access control
+
+Gruntwork Pipelines grants permissions by defining three [GitHub Teams](https://docs.github.com/en/organizations/organizing-members-into-teams/about-teams), which should map to three separate personas in your organization. Each team and its permissions are designed to grant the _least possible permissions_ to each individual (or machine user) in your organization for them to be able to perform changes to your infrastructure.
+
+### GitHub Team: infrastructure-administrators
+
+The `infrastructure-administrators` team is for engineers who likely work on the IaC codebase daily, and  have administrative AWS permissions. These users will also have permission to both read and write Pipelines configuration files, and the GitHub Actions Workflow files that are used to trigger Pipelines.
```
Comment (MoonMoon1919) [inline _docs-sources/pipelines/access-control/index.md:7]: When you say `pipelines configuration files` are you referring to the workflows in the infra-pipelines repository?

File: _docs-sources/pipelines/architecture/index.md (hunk 1 above)
Comment (MoonMoon1919) [reply _docs-sources/pipelines/architecture/index.md:9]: That is the plan eventually. We don't have that work scheduled for now.

File: _docs-sources/pipelines/architecture/index.md (hunk 7 above)
Comment (MoonMoon1919) [reply _docs-sources/pipelines/architecture/index.md:13]: Neither `NewAWSAccount` or `OpenTfModuleUpdate` are current infra change types in pipelines. You're likely looking for `ModuleUpdated`, `AccountRequested`, and `AccountAdded`

File: _docs-sources/pipelines/architecture/index.md [hunk 15]
```diff
@@ -0,0 +1,45 @@
...
+
+### The dispatcher
+
+The dispatcher takes as input an infrastructure change and a list of which pipelines actions to run with along with what parameters, and calls the executor to run these actions on the right infrastructure change.
```
Comment (MoonMoon1919) [inline _docs-sources/pipelines/architecture/index.md:17]: Currently it only takes a single action.

File: _docs-sources/pipelines/architecture/index.md [hunk 16]
```diff
@@ -0,0 +1,45 @@
...
+
+### The executor
+
+The executor executes an ordered list of pipelines actions against a specific pull request or git commit in response to a specific infrastructure. It then posts the status and output of these pipelines actions to the pull request, or takes other actions to ensure that the user has visibility into what happened.
```
Comment (MoonMoon1919) [inline _docs-sources/pipelines/architecture/index.md:21]: It doesn't respect the ordered list right now. We deferred that work in order to better support parallelization sooner.

File: _docs-sources/pipelines/architecture/index.md (hunk 4 above)
Comment (MoonMoon1919) [reply _docs-sources/pipelines/architecture/index.md:28]: Can do

File: _docs-sources/pipelines/how-it-works/index.md (hunk 3 above)
Comment (MoonMoon1919) [reply _docs-sources/pipelines/how-it-works/index.md:9]: I generally don't like bulleted lists in docs, they present information in a way that makes you quickly scan it rather than absorb the context through a well written paragraph.

File: _docs-sources/pipelines/supported-ci-systems/index.md (hunk 12 above)
Comment (MoonMoon1919) [reply _docs-sources/pipelines/supported-ci-systems/index.md:5]: the `pipelines` binary _is_ ci/cloud agnostic, but that currently only contains about 30-40% of the total logic for what is the current iteration of the pipelines product. I'd remove this for now and simple state that we currently support github actions, if you want another CI system, reach out to the product team.

File: _docs-sources/pipelines/pipelines-actions/index.md (hunk 11 above)
Comment (MoonMoon1919) [reply _docs-sources/pipelines/pipelines-actions/index.md:13]: +1. We shouldn't document behavior that doesn't exist yet.

File: _docs-sources/pipelines/initial-setup/index.md (hunk 5 above)
Comment (MoonMoon1919) [reply _docs-sources/pipelines/initial-setup/index.md:20]: Are you going to write this section?

File: _docs-sources/pipelines/core-concepts/index.md (hunk 8 above)
Comment (MoonMoon1919) [inline _docs-sources/pipelines/core-concepts/index.md:11]:
```suggestion
When you edit any infrastructure-as-code that needs to be "applied" to your cloud account (e.g. AWS or GCP), you are making an _infrastructure change_, or "infra-change" for short.
```

File: _docs-sources/pipelines/core-concepts/index.md (hunk 9 above)
Comment (MoonMoon1919) [inline _docs-sources/pipelines/core-concepts/index.md:13]: Currently we only support Terraform. None of these other tools are supported.

File: _docs-sources/pipelines/core-concepts/index.md [hunk 17]
```diff
@@ -0,0 +1,27 @@
...
+
+## Concepts
+
+### Infrastructure change
+
+When you edit any infrastructure-as-code that needs to be "applied" to your cloud account (e.g. AWS or GCP), you are making an _infrastructure change_. We sometime call these an "infra-change" for short.
+
+Infra-changes can involve updates to OpenTF (Terraform) code, Packer templates, Docker files, Kubernetes Helm charts, or any other type of file that represents a desired state of your infrastructure and that needs to be somehow applied. A classic example is changing some variables on an instance of a Terraform module. By changing these variables you are essentially requesting that some API calls eventually be made to AWS to update your infrastructure in some way (in this case, by running `terraform apply` at some point).
+
+Gruntwork Pipelines assumes that infra-changes are committed via git, usually by first opening a Pull Request. When you open a Pull Request, you are essentially "proposing" an infra-change.
```
Comment (MoonMoon1919) [inline _docs-sources/pipelines/core-concepts/index.md:15]: This is the only way gruntwork pipelines works, I think we need to reword this to remove weasel words like "assume" and "usually"

File: _docs-sources/pipelines/core-concepts/index.md [hunk 18]
```diff
@@ -0,0 +1,27 @@
...
+
+### Infrastructure change set
+
+Sometimes users create a Pull Request that changes more than one file at a time. And sometimes a change to a single file affects how multiple other files will be "applied" to your live cloud account. For example, many users of Terragrunt use a pattern known as "envcommon" as a way to specify a default set of values for modules.
```
Comment (MoonMoon1919) [inline _docs-sources/pipelines/core-concepts/index.md:19]: In other devops-foundations docs we've already switched this to be "module_defaults", so we should do the same here

File: _docs-sources/pipelines/core-concepts/index.md [hunk 19]
```diff
@@ -0,0 +1,27 @@
...
+
+When more than one infra-change is made at a time, we call this an _infrastructure-change set._ We sometimes call this an "infra-change set" for short.
```
Comment (MoonMoon1919) [inline _docs-sources/pipelines/core-concepts/index.md:21]: We always call it an infra change set, the number of infra changes is not consequential.

File: _docs-sources/pipelines/core-concepts/index.md (hunk 18 above)
Comment (MoonMoon1919) [inline _docs-sources/pipelines/core-concepts/index.md:19]:
```suggestion
An infrastructure change set (or infra-change set) is a collection of infra-changes determined by the changes to your IaC code.

There are two ways an infra-change set is generated -  Users creating a Pull Request that changes one or more environment specific files and when a default module is updated when using the [module_default pattern](../link-to-page). Changes to environment specific files result in an action running for that specific infrastructure unit. Changes to default modules result in actions being run in all environments that reference that default module.
```

I think we need to re-word this section. We don't explain what an infra change set is until we explain the types of changes that can be made, I think that's a bit backwards.

**Comment by josh-padnick** [comment]
@MoonMoon1919 @ellisonc Thanks for all your reviews and feedback. I'll share some feedback on some comments, but I don't want to be a blocker on this story, so please work with @ebeneliason to make the decisions you think are best. You do not need my approval or permission on anything here; you can make the call with Eben on what to launch based on your knowledge of Pipelines today and your own judgment.

File: _docs-sources/pipelines/how-it-works/index.md (hunk 3 above)
Comment (josh-padnick) [reply _docs-sources/pipelines/how-it-works/index.md:9]: The bulleted list here was intended as a placeholder, not the final product. But if there are stores in the backlog for this, that's probably a better place to do this work.

File: _docs-sources/pipelines/initial-setup/index.md (hunk 5 above)
Comment (josh-padnick) [reply _docs-sources/pipelines/initial-setup/index.md:20]: I actually wouldn't know how! But it's probably the single most actionable doc in the whole section, so we should have some plan for getting it written.

File: _docs-sources/pipelines/architecture/index.md (hunk 7 above)
Comment (josh-padnick) [reply _docs-sources/pipelines/architecture/index.md:13]: Roger on OpenTF. Roger on infra change types; please feel free to correct as needed.

File: _docs-sources/pipelines/pipelines-actions/index.md (hunk 11 above)
Comment (josh-padnick) [reply _docs-sources/pipelines/pipelines-actions/index.md:13]: I wanted to convey the concept of Pipelines Actions, but your (plural) instincts to keep all future promises out of docs is probably the right call.

File: _docs-sources/pipelines/supported-clouds/index.md (hunk 13 above)
Comment (josh-padnick) [reply _docs-sources/pipelines/supported-clouds/index.md:5]: Oh, that's a shame. We'll definitely want to support other clouds at some point. We do need to clarify that Pipelines only works with AWS today, but feel free to update the "designed to support any cloud" bit as needed.

File: _docs-sources/pipelines/access-control/index.md (hunk 14 above)
Comment (josh-padnick) [reply _docs-sources/pipelines/access-control/index.md:7]: I was making a bit of a leap here to include some future concept of configuration files that Pipelines admins will be able to edit. Feel free to updated as needed.

File: _docs-sources/pipelines/architecture/index.md (hunk 16 above)
Comment (josh-padnick) [reply _docs-sources/pipelines/architecture/index.md:21]: Roger, please update as needed

File: _docs-sources/pipelines/core-concepts/index.md (hunk 9 above)
Comment (josh-padnick) [reply _docs-sources/pipelines/core-concepts/index.md:13]: This is a conceptual point, though we should clarify explicitly that only Terraform files are supported today.

File: _docs-sources/pipelines/core-concepts/index.md (hunk 17 above)
Comment (josh-padnick) [reply _docs-sources/pipelines/core-concepts/index.md:15]: Roger; pleaes update as needed.

File: _docs-sources/pipelines/core-concepts/index.md (hunk 18 above)
Comment (josh-padnick) [reply _docs-sources/pipelines/core-concepts/index.md:19]: Roger; feel free to take the lead.

**Comment by josh-padnick** [comment]
@MoonMoon1919 @ellisonc Ok, just finished leaving all comments. Again, to be explicit, I won't plan to "review" any updates. So please run with things from here! My goal here was to share my thinking and insights, but without blocking you from getting this across the finish line.

File: _docs-sources/pipelines/how-it-works/index.md (hunk 3 above)
Comment (MoonMoon1919) [reply _docs-sources/pipelines/how-it-works/index.md:9]: ah i missed that, thanks!

File: _docs-sources/pipelines/architecture/index.md (hunk 15 above)
Comment (MoonMoon1919) [reply _docs-sources/pipelines/architecture/index.md:17]: > Also, plan and apply/destroy are two separate actions!

I think that's a bit misleading - we only support one _action_ per infra change. So we're going to plan or apply/destroy. I'll think of some phrasing.

**Comment by MoonMoon1919** [comment]
> was this meant to be compared against the devops-foundations branch?

@ellisonc I'm working off this because it contains all the core pipelines docs moves. Perhaps it should have gone into the devops foundations branch!

File: _docs-sources/pipelines/architecture/index.md (hunk 15 above)
Comment (josh-padnick) [reply _docs-sources/pipelines/architecture/index.md:17]: Oh, maybe you could help me build an accurate mental model there. In my mind an infra-change is "proposed" in a PR, then we run the first action on it (plan). Then when it's merged (for now, later we'll do some other trigger for applies), we run the apply/destroy action. How do you think about it?

File: _docs-sources/pipelines/access-control/index.md (hunk 14 above)
Comment (MoonMoon1919) [reply _docs-sources/pipelines/access-control/index.md:7]: Will do, thanks!

File: _docs-sources/pipelines/architecture/index.md (hunk 15 above)
Comment (MoonMoon1919) [reply _docs-sources/pipelines/architecture/index.md:17]: You're correct in the sense that a single infra change can have multiple actions run on it as a result of separate GitHub events (e.g., a PR open/sync or merging a PR to main). 

The point I was hoping to make here is that per invocation of pipelines there is only one action (either a plan, apply, or destroy) per infra change.

**Comment by MoonMoon1919** [comment]
I'm going to keep this open for posterity so we can work from it for future stories, but for now I'm going to fold the initial changes for this story here: https://github.com/gruntwork-io/docs/pull/1099

Feedback from this PR addressed.

File: _docs-sources/pipelines/architecture/index.md (hunk 15 above)
Comment (josh-padnick) [reply _docs-sources/pipelines/architecture/index.md:17]: Oh, I see. That makes sense. 👍 

I suppose in the future, a single git event could still trigger more than one action (e.g. plan, then cost estimate), but yes, for now, each "run" supports one pipelines action only.

**Comment by MoonMoon1919** [comment]
Closing this because there are _many_ merge conflicts at this point. I believe we've folded everything in here.

**Comment by josh-padnick** [comment]
@MoonMoon1919 FYI, I referenced this PR a bunch of times to add content in other PRs, so I don't think we're losing anything material by closing this.
